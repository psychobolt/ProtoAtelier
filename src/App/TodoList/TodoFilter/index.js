export { default as Filters } from './TodoFilter.filters';
export { default } from './TodoFilter.component';
