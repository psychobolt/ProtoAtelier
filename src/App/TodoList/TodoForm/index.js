export * from './TodoForm.actions';
export { default } from './TodoForm.container';
