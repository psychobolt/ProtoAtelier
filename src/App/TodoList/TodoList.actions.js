export const Actions = {
  TOGGLE_TODO: 'toggleTodo',
  UNDO_TODO: 'undoTodo',
  REDO_TODO: 'redoTodo',
};

export const toggleTodo = id => ({
  type: Actions.TOGGLE_TODO,
  payload: { id },
});

export const undoTodo = () => ({ type: Actions.UNDO_TODO });
export const redoTodo = () => ({ type: Actions.REDO_TODO });
