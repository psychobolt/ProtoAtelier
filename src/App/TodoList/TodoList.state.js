// @flow
import type { Todo } from './TodoItem';

export type TodoListState = {
  todos: Todo[]
};

export default [];
