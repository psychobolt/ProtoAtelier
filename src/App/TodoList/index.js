export { default as TodoFilter, Filters } from './TodoFilter';
export { default as TodoForm } from './TodoForm';
export { default } from './TodoList.container';
