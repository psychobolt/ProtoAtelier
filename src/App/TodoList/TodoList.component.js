// @flow
import React from 'react';
import styled from 'styled-components';

import TodoItem, { type Todo } from './TodoItem';
import * as styles from './TodoList.style';

type Props = {
  todos: Todo[],
  onTodoClick: (id: number) => void,
  children: any,
  className: string,
}

const List = styled.ul`${styles.ul}`;

const Item = styled(TodoItem)`${styles.li}`;

export default ({ todos, onTodoClick, children, className }: Props) => (
  <x-card class={className}>
    {children}
    <List>
      {todos.length ? todos.map(todo => (
        <li key={todo.id}>
          <Item
            {...todo}
            onClick={() => onTodoClick(todo.id)}
          />
        </li>
      )) : (
        <x-label>
          {'No items'}
        </x-label>
      )}
    </List>
  </x-card>
);
