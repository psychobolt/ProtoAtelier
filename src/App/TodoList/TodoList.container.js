// @flow
import type { Dispatch } from 'redux';
import { connect } from 'react-redux';

import TodoList from './TodoList.component';
import { toggleTodo } from './TodoList.actions';
import { getVisibleTodos } from './TodoList.selectors';
import type { TodoListState } from './TodoList.state';

type Props = {
  filter: string
};

export const mapStateToProps = (state: TodoListState, props: Props) => ({
  todos: getVisibleTodos(state, props),
});

const mapDispatchToProps = (dispatch: Dispatch<any>) => ({
  onTodoClick: id => {
    dispatch(toggleTodo(id));
  },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(TodoList);
