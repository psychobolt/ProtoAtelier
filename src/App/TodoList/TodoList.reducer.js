import initialState from './TodoList.state';
import { Actions as TodoListActions } from './TodoList.actions';
import { Actions as TodoFormActions } from './TodoForm';

export default (state = initialState, action) => {
  switch (action.type) {
    case TodoFormActions.ADD_TODO:
      return [
        ...state,
        {
          id: action.todo.id,
          text: action.todo.text,
          completed: false,
        },
      ];
    case TodoListActions.TOGGLE_TODO:
      return state.map(todo =>
        (todo.id === action.todoId ? { ...todo, completed: !todo.completed } : todo));
    default:
      return state;
  }
};
