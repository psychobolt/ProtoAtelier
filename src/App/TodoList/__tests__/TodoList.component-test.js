import React from 'react';
import { shallow, mount } from 'enzyme';

import TodoList from '../TodoList.component';

describe('components <TodoList />', () => {
  it('TodoList should render without items', () => {
    const props = {
      todos: [],
      onTodoClick: jest.fn(),
    };
    shallow(<TodoList {...props} />);
  });

  it('TodoList should render without crashing -- completed', () => {
    const props = {
      todos: [{
        id: 1,
        completed: true,
        text: 'TodoItem',
      }],
      onTodoClick: jest.fn(),
    };
    shallow(<TodoList {...props} />);
  });

  it('TodoList should render without crashing -- not completed', () => {
    const props = {
      todos: [{
        id: 1,
        completed: false,
        text: 'TodoItem',
      }],
      onTodoClick: jest.fn(),
    };
    shallow(<TodoList {...props} />);
  });

  it('TodoList should call callback on item click', () => {
    const props = {
      todos: [{
        id: 1,
        completed: false,
        text: 'TodoItem',
      }],
      onTodoClick: jest.fn(),
    };
    const wrapper = mount(<TodoList {...props} />);
    wrapper.find('x-checkbox').first().simulate('click');
    expect(props.onTodoClick.mock.calls.length).toBe(1);
  });
});
