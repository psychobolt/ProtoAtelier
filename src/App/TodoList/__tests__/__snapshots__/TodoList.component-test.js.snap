// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <TodoList /> TodoList should render correctly -- completed 1`] = `
<x-card
  style={Object {}}
>
  <ul
    style={
      Object {
        "listStyleType": "none",
      }
    }
  >
    <li
      key="1"
    >
      <TodoItem
        completed={true}
        id={1}
        onClick={[Function]}
        style={
          Object {
            "marginBottom": "5px",
          }
        }
        text="TodoItem"
      />
    </li>
  </ul>
</x-card>
`;

exports[`components <TodoList /> TodoList should render correctly -- not completed 1`] = `
<x-card
  style={Object {}}
>
  <ul
    style={
      Object {
        "listStyleType": "none",
      }
    }
  >
    <li
      key="1"
    >
      <TodoItem
        completed={false}
        id={1}
        onClick={[Function]}
        style={
          Object {
            "marginBottom": "5px",
          }
        }
        text="TodoItem"
      />
    </li>
  </ul>
</x-card>
`;
